### DISCUSSION TITLE
Kubernetes: Gluetun as a sidecar is constantly restarting

### DISCUSSION THREAD

**Discussion by renannprado** [qdm12/gluetun#2006] [Q&A]
Hello dear community.

I'm facing an issue where the VPN sidecar container is constantly restarting the VPN connection.
While it's up and running, the VPN works just fine, but it's restarting too often.

At the moment, I'm clueless on how to proceed with the debugging, that's why I'm creating this thread.

My setup: [k0s](https://k0sproject.io/) with [cilium](https://cilium.io/), no kube-proxy. [Here's the guide](https://docs.cilium.io/en/stable/installation/k0s/#k0s-install).

I've also tried with [calico](https://www.tigera.io/project-calico/) as my CNI, but the exact same issue is happening. Actually it was the first one I tried, but after seeing the below thread (1), I've tried to switch to cilium.

Here's the relevant part of the deployment:

```yaml
- env:
  - name: VPN_SERVICE_PROVIDER
    value: private internet access
  - name: OPENVPN_USER
    value: secret
  - name: OPENVPN_PASSWORD
    value: secret
  - name: SERVER_REGIONS
    value: Belgium
  image: qmcgaw/gluetun:pr-1742@sha256:d268ea559332437c1a9b2f5a63813df7908e22ed1b81aafeb1842d5abd7ee563
  imagePullPolicy: IfNotPresent
  name: vpn
  securityContext:
    capabilities:
      add:
      - NET_ADMIN
```

<details>

<summary>Logs</summary>

```
========================================
========================================
=============== gluetun ================
========================================
=========== Made with ❤️ by ============
======= https://github.com/qdm12 =======
========================================
========================================

Running version pr-1742 built on 2023-11-01T14:30:04.842Z (commit a40507f)

🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
🐛 Bug? https://github.com/qdm12/gluetun/issues/new
✨ New feature? https://github.com/qdm12/gluetun/issues/new
☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2023-12-15T09:50:30Z INFO [routing] default route found: interface eth0, gateway 10.0.0.46, assigned IP 10.0.0.110 and family v4
2023-12-15T09:50:30Z INFO [routing] local ethernet link found: eth0
2023-12-15T09:50:30Z INFO [routing] local ipnet found: 10.0.0.46/32
2023-12-15T09:50:30Z INFO [routing] local ipnet found: fe80::/64
2023-12-15T09:50:30Z INFO [firewall] enabling...
2023-12-15T09:50:30Z INFO [firewall] enabled successfully
2023-12-15T09:50:31Z INFO [storage] creating /gluetun/servers.json with 17689 hardcoded servers
2023-12-15T09:50:31Z INFO Alpine version: 3.18.4
2023-12-15T09:50:31Z INFO OpenVPN 2.5 version: 2.5.8
2023-12-15T09:50:31Z INFO OpenVPN 2.6 version: 2.6.5
2023-12-15T09:50:31Z INFO IPtables version: v1.8.9
2023-12-15T09:50:31Z INFO Settings summary:
├── VPN settings:
|   ├── VPN provider settings:
|   |   ├── Name: private internet access
|   |   └── Server selection settings:
|   |       ├── VPN type: openvpn
|   |       ├── Regions: belgium
|   |       └── OpenVPN server selection settings:
|   |           ├── Protocol: UDP
|   |           └── Private Internet Access encryption preset: strong
|   └── OpenVPN settings:
|       ├── OpenVPN version: 2.5
|       ├── User: [set]
|       ├── Password: [set]
|       ├── Private Internet Access encryption preset: strong
|       ├── Network interface: tun0
|       ├── Run OpenVPN as: root
|       └── Verbosity level: 1
├── DNS settings:
|   ├── Keep existing nameserver(s): no
|   ├── DNS server address to use: 127.0.0.1
|   └── DNS over TLS settings:
|       ├── Enabled: yes
|       ├── Update period: every 24h0m0s
|       ├── Authoritative servers:
|       |   └── cloudflare
|       ├── Caching: yes
|       ├── IPv6: no
|       └── DNS filtering settings:
|           ├── Block malicious: yes
|           ├── Block ads: no
|           ├── Block surveillance: no
|           └── Blocked IP networks:
|               ├── 127.0.0.1/8
|               ├── 10.0.0.0/8
|               ├── 172.16.0.0/12
|               ├── 192.168.0.0/16
|               ├── 169.254.0.0/16
|               ├── ::1/128
|               ├── fc00::/7
|               ├── fe80::/10
|               ├── ::ffff:127.0.0.1/104
|               ├── ::ffff:10.0.0.0/104
|               ├── ::ffff:169.254.0.0/112
|               ├── ::ffff:172.16.0.0/108
|               └── ::ffff:192.168.0.0/112
├── Firewall settings:
|   └── Enabled: yes
├── Log settings:
|   └── Log level: INFO
├── Health settings:
|   ├── Server listening address: 127.0.0.1:9999
|   ├── Target address: cloudflare.com:443
|   ├── Duration to wait after success: 5s
|   ├── Read header timeout: 100ms
|   ├── Read timeout: 500ms
|   └── VPN wait durations:
|       ├── Initial duration: 6s
|       └── Additional duration: 5s
├── Shadowsocks server settings:
|   └── Enabled: no
├── HTTP proxy settings:
|   └── Enabled: no
├── Control server settings:
|   ├── Listening address: :8000
|   └── Logging: yes
├── OS Alpine settings:
|   ├── Process UID: 1000
|   └── Process GID: 1000
├── Public IP settings:
|   ├── Fetching: every 12h0m0s
|   └── IP file path: /tmp/gluetun/ip
└── Version settings:
    └── Enabled: yes
2023-12-15T09:50:31Z INFO [routing] default route found: interface eth0, gateway 10.0.0.46, assigned IP 10.0.0.110 and family v4
2023-12-15T09:50:31Z INFO [routing] adding route for 0.0.0.0/0
2023-12-15T09:50:31Z INFO [firewall] setting allowed subnets...
2023-12-15T09:50:31Z INFO [routing] default route found: interface eth0, gateway 10.0.0.46, assigned IP 10.0.0.110 and family v4
2023-12-15T09:50:31Z INFO TUN device is not available: open /dev/net/tun: no such file or directory; creating it...
2023-12-15T09:50:31Z INFO [http server] http server listening on [::]:8000
2023-12-15T09:50:31Z INFO [dns] using plaintext DNS at address 1.1.1.1
2023-12-15T09:50:31Z INFO [healthcheck] listening on 127.0.0.1:9999
2023-12-15T09:50:31Z INFO [firewall] allowing VPN connection...
2023-12-15T09:50:31Z INFO [openvpn] OpenVPN 2.5.8 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Nov  2 2022
2023-12-15T09:50:31Z INFO [openvpn] library versions: OpenSSL 3.1.4 24 Oct 2023, LZO 2.10
2023-12-15T09:50:31Z INFO [openvpn] CRL: loaded 1 CRLs from file -----BEGIN X509 CRL-----
2023-12-15T09:50:31Z INFO [openvpn] MIIDWDCCAUAwDQYJKoZIhvcNAQENBQAwgegxCzAJBgNVBAYTAlVTMQswCQYDVQQIEwJDQTETMBEGA1UEBxMKTG9zQW5nZWxlczEgMB4GA1UEChMXUHJpdmF0ZSBJbnRlcm5ldCBBY2Nlc3MxIDAeBgNVBAsTF1ByaXZhdGUgSW50ZXJuZXQgQWNjZXNzMSAwHgYDVQQDExdQcml2YXRlIEludGVybmV0IEFjY2VzczEgMB4GA1UEKRMXUHJpdmF0ZSBJbnRlcm5ldCBBY2Nlc3MxLzAtBgkqhkiG9w0BCQEWIHNlY3VyZUBwcml2YXRlaW50ZXJuZXRhY2Nlc3MuY29tFw0xNjA3MDgxOTAwNDZaFw0zNjA3MDMxOTAwNDZaMCYwEQIBARcMMTYwNzA4MTkwMDQ2MBECAQYXDDE2MDcwODE5MDA0NjANBgkqhkiG9w0BAQ0FAAOCAgEAppFfEpGsasjB1QgJcosGpzbf2kfRhM84o2TlqY1ua+Gi5TMdKydA3LJcNTjlI9a0TYAJfeRX5IkpoglSUuHuJgXhP3nEvX10mjXDpcu/YvM8TdE5JV2+EGqZ80kFtBeOq94WcpiVKFTR4fO+VkOK9zwspFfb1cNs9rHvgJ1QMkRUF8PpLN6AkntHY0+6DnigtSaKqldqjKTDTv2OeH3nPoh80SGrt0oCOmYKfWTJGpggMGKvIdvU3vH9+EuILZKKIskt+1dwdfA5Bkz1GLmiQG7+9ZZBQUjBG9Dos4hfX/rwJ3eU8oUIm4WoTz9rb71SOEuUUjP5NPy9HNx2vx+cVvLsTF4ZDZaUztW9o9JmIURDtbeyqxuHN3prlPWB6aj73IIm2dsDQvs3XXwRIxs8NwLbJ6CyEuvEOVCskdM8rdADWx1J0lRNlOJ0Z8ieLLEmYAA834VN1SboB6wJIAPxQU3rcBhXqO9y8aa2oRMg8NxZ5gr+PnKVMqag1x0IxbIgLxtkXQvxXxQHEMSODzvcOfK/nBRBsqTj30P+R87sU8titOoxNeRnBDRNhdEy/QGAqGh62ShPpQUCJdnKRiRTjnil9hMQHevoSuFKeEMO30FQL7BZyo37GFU+q1WPCplVZgCP9hC8Rn5K2+f6KLFo5bhtowSmu+GY1yZtg+RTtsA=
2023-12-15T09:50:31Z INFO [openvpn] -----END X509 CRL-----
2023-12-15T09:50:31Z INFO [openvpn] TCP/UDP: Preserving recently used remote address: [AF_INET]181.214.218.42:1197
2023-12-15T09:50:31Z INFO [openvpn] UDP link local: (not bound)
2023-12-15T09:50:31Z INFO [openvpn] UDP link remote: [AF_INET]181.214.218.42:1197
2023-12-15T09:50:35Z WARN [openvpn] 'link-mtu' is used inconsistently, local='link-mtu 1569', remote='link-mtu 1554'
2023-12-15T09:50:35Z WARN [openvpn] 'keysize' is used inconsistently, local='keysize 256', remote='keysize 128'
2023-12-15T09:50:35Z WARN [openvpn] 'comp-lzo' is present in remote config but missing in local config, remote='comp-lzo'
2023-12-15T09:50:35Z INFO [openvpn] [brussels418] Peer Connection Initiated with [AF_INET]181.214.218.42:1197
2023-12-15T09:50:35Z INFO [openvpn] sitnl_send: rtnl: generic error (-101): Network unreachable
2023-12-15T09:50:35Z INFO [openvpn] TUN/TAP device tun0 opened
2023-12-15T09:50:35Z INFO [openvpn] /sbin/ip link set dev tun0 up mtu 1500
2023-12-15T09:50:35Z INFO [openvpn] /sbin/ip link set dev tun0 up
2023-12-15T09:50:35Z INFO [openvpn] /sbin/ip addr add dev tun0 10.11.110.181/24
2023-12-15T09:50:35Z WARN [openvpn] OpenVPN was configured to add an IPv6 route. However, no IPv6 has been configured for tun0, therefore the route installation may fail or may not work as expected.
2023-12-15T09:50:35Z INFO [openvpn] add_route_ipv6(2000::/3 -> :: metric -1) dev tun0
2023-12-15T09:50:35Z INFO [openvpn] UID set to nonrootuser
2023-12-15T09:50:35Z INFO [openvpn] Initialization Sequence Completed
2023-12-15T09:50:35Z INFO [dns] downloading hostnames and IP block lists
2023-12-15T09:50:36Z INFO [healthcheck] healthy!
2023-12-15T09:50:44Z INFO [healthcheck] unhealthy: dialing: dial tcp4: lookup cloudflare.com: i/o timeout
2023-12-15T09:50:52Z INFO [healthcheck] program has been unhealthy for 6s: restarting VPN (see https://github.com/qdm12/gluetun-wiki/blob/main/faq/healthcheck.md)
2023-12-15T09:50:52Z INFO [vpn] stopping
2023-12-15T09:50:52Z ERROR [vpn] cannot get version information: Get "https://api.github.com/repos/qdm12/gluetun/releases": context canceled
2023-12-15T09:50:52Z INFO [vpn] starting
2023-12-15T09:50:52Z INFO [firewall] allowing VPN connection...
2023-12-15T09:50:52Z INFO [openvpn] OpenVPN 2.5.8 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Nov  2 2022
2023-12-15T09:50:52Z INFO [openvpn] library versions: OpenSSL 3.1.4 24 Oct 2023, LZO 2.10
2023-12-15T09:50:52Z INFO [openvpn] CRL: loaded 1 CRLs from file -----BEGIN X509 CRL-----
2023-12-15T09:50:52Z INFO [openvpn] MIIDWDCCAUAwDQYJKoZIhvcNAQENBQAwgegxCzAJBgNVBAYTAlVTMQswCQYDVQQIEwJDQTETMBEGA1UEBxMKTG9zQW5nZWxlczEgMB4GA1UEChMXUHJpdmF0ZSBJbnRlcm5ldCBBY2Nlc3MxIDAeBgNVBAsTF1ByaXZhdGUgSW50ZXJuZXQgQWNjZXNzMSAwHgYDVQQDExdQcml2YXRlIEludGVybmV0IEFjY2VzczEgMB4GA1UEKRMXUHJpdmF0ZSBJbnRlcm5ldCBBY2Nlc3MxLzAtBgkqhkiG9w0BCQEWIHNlY3VyZUBwcml2YXRlaW50ZXJuZXRhY2Nlc3MuY29tFw0xNjA3MDgxOTAwNDZaFw0zNjA3MDMxOTAwNDZaMCYwEQIBARcMMTYwNzA4MTkwMDQ2MBECAQYXDDE2MDcwODE5MDA0NjANBgkqhkiG9w0BAQ0FAAOCAgEAppFfEpGsasjB1QgJcosGpzbf2kfRhM84o2TlqY1ua+Gi5TMdKydA3LJcNTjlI9a0TYAJfeRX5IkpoglSUuHuJgXhP3nEvX10mjXDpcu/YvM8TdE5JV2+EGqZ80kFtBeOq94WcpiVKFTR4fO+VkOK9zwspFfb1cNs9rHvgJ1QMkRUF8PpLN6AkntHY0+6DnigtSaKqldqjKTDTv2OeH3nPoh80SGrt0oCOmYKfWTJGpggMGKvIdvU3vH9+EuILZKKIskt+1dwdfA5Bkz1GLmiQG7+9ZZBQUjBG9Dos4hfX/rwJ3eU8oUIm4WoTz9rb71SOEuUUjP5NPy9HNx2vx+cVvLsTF4ZDZaUztW9o9JmIURDtbeyqxuHN3prlPWB6aj73IIm2dsDQvs3XXwRIxs8NwLbJ6CyEuvEOVCskdM8rdADWx1J0lRNlOJ0Z8ieLLEmYAA834VN1SboB6wJIAPxQU3rcBhXqO9y8aa2oRMg8NxZ5gr+PnKVMqag1x0IxbIgLxtkXQvxXxQHEMSODzvcOfK/nBRBsqTj30P+R87sU8titOoxNeRnBDRNhdEy/QGAqGh62ShPpQUCJdnKRiRTjnil9hMQHevoSuFKeEMO30FQL7BZyo37GFU+q1WPCplVZgCP9hC8Rn5K2+f6KLFo5bhtowSmu+GY1yZtg+RTtsA=
2023-12-15T09:50:52Z INFO [openvpn] -----END X509 CRL-----
2023-12-15T09:50:52Z INFO [openvpn] TCP/UDP: Preserving recently used remote address: [AF_INET]181.214.218.32:1197
2023-12-15T09:50:52Z INFO [openvpn] UDP link local: (not bound)
2023-12-15T09:50:52Z INFO [openvpn] UDP link remote: [AF_INET]181.214.218.32:1197
2023-12-15T09:50:52Z WARN [openvpn] 'link-mtu' is used inconsistently, local='link-mtu 1569', remote='link-mtu 1554'
2023-12-15T09:50:52Z WARN [openvpn] 'keysize' is used inconsistently, local='keysize 256', remote='keysize 128'
2023-12-15T09:50:52Z WARN [openvpn] 'comp-lzo' is present in remote config but missing in local config, remote='comp-lzo'
2023-12-15T09:50:52Z INFO [openvpn] [brussels418] Peer Connection Initiated with [AF_INET]181.214.218.32:1197
2023-12-15T09:50:52Z INFO [openvpn] sitnl_send: rtnl: generic error (-101): Network unreachable
2023-12-15T09:50:52Z INFO [openvpn] TUN/TAP device tun0 opened
2023-12-15T09:50:52Z INFO [openvpn] /sbin/ip link set dev tun0 up mtu 1500
2023-12-15T09:50:52Z INFO [openvpn] /sbin/ip link set dev tun0 up
2023-12-15T09:50:52Z INFO [openvpn] /sbin/ip addr add dev tun0 10.1.110.197/24
2023-12-15T09:50:52Z WARN [openvpn] OpenVPN was configured to add an IPv6 route. However, no IPv6 has been configured for tun0, therefore the route installation may fail or may not work as expected.
2023-12-15T09:50:52Z INFO [openvpn] add_route_ipv6(2000::/3 -> :: metric -1) dev tun0
2023-12-15T09:50:52Z INFO [openvpn] UID set to nonrootuser
2023-12-15T09:50:52Z INFO [openvpn] Initialization Sequence Completed
2023-12-15T09:50:53Z INFO [healthcheck] healthy!
2023-12-15T09:51:01Z INFO [healthcheck] unhealthy: dialing: dial tcp4: lookup cloudflare.com: i/o timeout
2023-12-15T09:51:03Z INFO [ip getter] Public IP address is 181.214.218.32 (Belgium, Brussels Capital, Brussels)
2023-12-15T09:51:04Z INFO [ip getter] Public IP address is 181.214.218.32 (Belgium, Brussels Capital, Brussels)
2023-12-15T09:51:09Z INFO [healthcheck] program has been unhealthy for 6s: restarting VPN (see https://github.com/qdm12/gluetun-wiki/blob/main/faq/healthcheck.md)
2023-12-15T09:51:09Z INFO [vpn] stopping
2023-12-15T09:51:09Z INFO [vpn] starting
2023-12-15T09:51:09Z INFO [firewall] allowing VPN connection...
2023-12-15T09:51:09Z INFO [openvpn] OpenVPN 2.5.8 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Nov  2 2022
2023-12-15T09:51:09Z INFO [openvpn] library versions: OpenSSL 3.1.4 24 Oct 2023, LZO 2.10
2023-12-15T09:51:09Z INFO [openvpn] CRL: loaded 1 CRLs from file -----BEGIN X509 CRL-----
2023-12-15T09:51:09Z INFO [openvpn] MIIDWDCCAUAwDQYJKoZIhvcNAQENBQAwgegxCzAJBgNVBAYTAlVTMQswCQYDVQQIEwJDQTETMBEGA1UEBxMKTG9zQW5nZWxlczEgMB4GA1UEChMXUHJpdmF0ZSBJbnRlcm5ldCBBY2Nlc3MxIDAeBgNVBAsTF1ByaXZhdGUgSW50ZXJuZXQgQWNjZXNzMSAwHgYDVQQDExdQcml2YXRlIEludGVybmV0IEFjY2VzczEgMB4GA1UEKRMXUHJpdmF0ZSBJbnRlcm5ldCBBY2Nlc3MxLzAtBgkqhkiG9w0BCQEWIHNlY3VyZUBwcml2YXRlaW50ZXJuZXRhY2Nlc3MuY29tFw0xNjA3MDgxOTAwNDZaFw0zNjA3MDMxOTAwNDZaMCYwEQIBARcMMTYwNzA4MTkwMDQ2MBECAQYXDDE2MDcwODE5MDA0NjANBgkqhkiG9w0BAQ0FAAOCAgEAppFfEpGsasjB1QgJcosGpzbf2kfRhM84o2TlqY1ua+Gi5TMdKydA3LJcNTjlI9a0TYAJfeRX5IkpoglSUuHuJgXhP3nEvX10mjXDpcu/YvM8TdE5JV2+EGqZ80kFtBeOq94WcpiVKFTR4fO+VkOK9zwspFfb1cNs9rHvgJ1QMkRUF8PpLN6AkntHY0+6DnigtSaKqldqjKTDTv2OeH3nPoh80SGrt0oCOmYKfWTJGpggMGKvIdvU3vH9+EuILZKKIskt+1dwdfA5Bkz1GLmiQG7+9ZZBQUjBG9Dos4hfX/rwJ3eU8oUIm4WoTz9rb71SOEuUUjP5NPy9HNx2vx+cVvLsTF4ZDZaUztW9o9JmIURDtbeyqxuHN3prlPWB6aj73IIm2dsDQvs3XXwRIxs8NwLbJ6CyEuvEOVCskdM8rdADWx1J0lRNlOJ0Z8ieLLEmYAA834VN1SboB6wJIAPxQU3rcBhXqO9y8aa2oRMg8NxZ5gr+PnKVMqag1x0IxbIgLxtkXQvxXxQHEMSODzvcOfK/nBRBsqTj30P+R87sU8titOoxNeRnBDRNhdEy/QGAqGh62ShPpQUCJdnKRiRTjnil9hMQHevoSuFKeEMO30FQL7BZyo37GFU+q1WPCplVZgCP9hC8Rn5K2+f6KLFo5bhtowSmu+GY1yZtg+RTtsA=
2023-12-15T09:51:09Z INFO [openvpn] -----END X509 CRL-----
2023-12-15T09:51:09Z INFO [openvpn] TCP/UDP: Preserving recently used remote address: [AF_INET]181.214.218.35:1197
2023-12-15T09:51:09Z INFO [openvpn] UDP link local: (not bound)
2023-12-15T09:51:09Z INFO [openvpn] UDP link remote: [AF_INET]181.214.218.35:1197
2023-12-15T09:51:15Z WARN [openvpn] 'link-mtu' is used inconsistently, local='link-mtu 1569', remote='link-mtu 1554'
2023-12-15T09:51:15Z WARN [openvpn] 'keysize' is used inconsistently, local='keysize 256', remote='keysize 128'
2023-12-15T09:51:15Z WARN [openvpn] 'comp-lzo' is present in remote config but missing in local config, remote='comp-lzo'
2023-12-15T09:51:15Z INFO [openvpn] [brussels418] Peer Connection Initiated with [AF_INET]181.214.218.35:1197
2023-12-15T09:51:15Z INFO [openvpn] sitnl_send: rtnl: generic error (-101): Network unreachable
2023-12-15T09:51:15Z INFO [openvpn] TUN/TAP device tun0 opened
2023-12-15T09:51:15Z INFO [openvpn] /sbin/ip link set dev tun0 up mtu 1500
2023-12-15T09:51:15Z INFO [openvpn] /sbin/ip link set dev tun0 up
2023-12-15T09:51:15Z INFO [openvpn] /sbin/ip addr add dev tun0 10.4.110.201/24
2023-12-15T09:51:15Z WARN [openvpn] OpenVPN was configured to add an IPv6 route. However, no IPv6 has been configured for tun0, therefore the route installation may fail or may not work as expected.
2023-12-15T09:51:15Z INFO [openvpn] add_route_ipv6(2000::/3 -> :: metric -1) dev tun0
2023-12-15T09:51:15Z INFO [openvpn] UID set to nonrootuser
2023-12-15T09:51:15Z INFO [openvpn] Initialization Sequence Completed
2023-12-15T09:51:20Z INFO [healthcheck] program has been unhealthy for 11s: restarting VPN (see https://github.com/qdm12/gluetun-wiki/blob/main/faq/healthcheck.md)
2023-12-15T09:51:20Z INFO [vpn] stopping
2023-12-15T09:51:20Z INFO [vpn] starting
2023-12-15T09:51:20Z INFO [firewall] allowing VPN connection...
```

</details>

I've followed some examples provided in the below thread provided by, it doesn't look like I'm doing something too differenly.

@Noah-Huppert I'm tagging you here since you seem to have good experience with similar setup so far, I hope that's fine.

Edit 16/12/2023:

I just wanted to share that if I use the docker instead of kubernetes in the same machine, the restarts don't happen.

(1) https://github.com/qdm12/gluetun/discussions/1762#discussioncomment-7824605

**Comment by ezekieldas** [comment]
I'm not so knowledgeable with k8s or k0s but looking at your log it looks there's a consecutive healthcheck connection that isn't completing. For troubleshooting you might extend the healthcheck interval and see what's happening in the container. For example, once the connection is up, can you `curl cloudflare.com:443`

Also, I'm reminded of something on this page re: `FIREWALL_INPUT_PORTS`: 

https://github.com/qdm12/gluetun-wiki/blob/29e3dee5024d82218e3f170b32f027102ff84aab/setup/options/firewall.md

**Reply by renannprado** [reply]
@ezekieldas I'm not sure if I understand what's the purpose of `FIREWALL_INPUT_PORTS` exactly.

Does that mean that those ports are allowed to escape the VPN?

**Reply by ezekieldas** [reply]
Check the wiki: https://github.com/qdm12/gluetun-wiki/blob/main/setup/options/firewall.md 

Also, check issues. Here's something that looks similar: https://github.com/qdm12/gluetun/issues/254

**Reply by renannprado** [reply]
@ezekieldas I've tried to put ports `80,443` in `FIREWALL_INPUT_PORTS`, but nothing changed 😞 .

And I saw the documentation, but I still don't understand what `Comma separated list of ports to allow through the default interface. This seems needed for Kubernetes sidecars.` means. What is the default interface? Is it `tun0`? Is it supposed to allow ports that go through the VPN or that _don't_ go through the VPN? I don't get it. Once this is clear to me, I'll send a PR to make the doc clearer.

I didn't get the chance to extend the healthcheck interval, but I noticed from within both sidecar container and the gluetun container itself, the connection really drops eventually. But it definitely is healthy for some time.

Since the gluetun with the same config was up and running just fine within docker, I'm starting to wonder if the underlying container runtime (`containerd`) could be causing the issues, so yet another thing to verify in case I don't manage to make it work.

About the #254, I'll have a look as soon as I can and I'll come back to you. Thanks for the help!

**Reply by renannprado** [reply]
After changing some healthcheck options, It seems to be related to the long time that the requests is having to wait.

The response time of ip.me and cloudflare vary a lot with the VPN turned on and the default configuration is not enough to consider the vpn healthy.

<img width="856" alt="image" src="https://github.com/qdm12/gluetun/assets/2359543/742d63d7-e4cc-4c9b-a8f1-de972a0ae497">

Of course, connection will be slow with VPN, but I wonder if there could be something else wrong as well.

Next step is to check (as in #254) whether I can access the service via the ingress.

TL;DR setting `HEALTH_VPN_DURATION_INITIAL`, `HEALTH_VPN_DURATION_ADDITION` and `HEALTH_SUCCESS_WAIT_DURATION` to `1m` helped me.

**Reply by ezekieldas** [reply]
> What is the default interface? Is it tun0? Is it supposed to allow ports that go through the VPN or that don't go through the VPN?

Good question. I think you can determine how this is being applied by using `FIREWALL_DEBUG: on` or by accessing the container and using `iptables`. 

> Since the gluetun with the same config was up and running just fine within docker, I'm starting to wonder if the underlying container runtime (containerd) could be causing the issues...

Is there a super simple container you could sidecar and check network? Those response times don't look good and increasing the related parameters is enabling bad behavior.

**Reply by renannprado** [reply]
@ezekieldas I'll figure out the answer to the first question later then.

How can I debug any further here?

I've just spin up gluetun via docker compose and did a `speedtest`:

<img width="859" alt="image" src="https://github.com/qdm12/gluetun/assets/2359543/aeffed77-bdd0-4c74-b55c-cca63804ae5d">

So the behavior is really inside of kubernetes, but not with docker alone. 

Should I try to run wireshark to inspect the packets or do you have a better/easier idea?

Edit:

I have a debug sidecar container as well, so that's not a problem. I just need to know how to proceed any further.

**Reply by ezekieldas** [reply]
> I have a debug sidecar container as well

I'd recommend using this to determine latency / port issues. You could also experiment with a WG (or openvpn) connection in the debug sidecar without using gluetun. Something [like this](https://www.digitalocean.com/community/tutorials/how-to-set-up-wireguard-on-ubuntu-20-04) for example.

**Reply by renannprado** [reply]
@ezekieldas I gave up for now, it has consumed to much time already.
I'll either install the VPN in a VM or docker and move into kubernetes in the future or if someone else has gone through similar issues.

Thanks for the help so far and happy new year 😄
